Update learn more link in Sign in with Google toggle.

benbowler opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #9734

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

benbowler commented:
Note, as of creating this PR the URL links to 404 but I assume the documentation is yet to be published.

10upsimon reviewed (approved):
LGTM ✅

aaemnnosttv reviewed (commented):
Thanks @benbowler – this will need updating as this issue was somewhat incompletely defined. I've gone ahead and updated the AC + IB to be more clear for consistency regarding our approach to these kinds of links.

aaemnnosttv reviewed (changes requested):
Sending back for a quick follow-up.

benbowler commented:
Hey @aaemnnosttv, I had to update the `getDocumentationLinkURL` selector to support hash properties as well to meet the AC. Updated and back to CR.

aaemnnosttv commented on assets/js/googlesitekit/datastore/site/urls.js:
```diff
@@ -112,20 +112,33 @@ export const selectors = {
 	 * @since 1.80.0
+	 * @since n.e.x.t Added the `hasProperty` argument.
 	 *
-	 * @param {Object} state Data store's state.
-	 * @param {string} slug  The slug of the documentation page.
+	 * @param {Object}  state       Data store's state.
+	 * @param {string}  slug        The slug of the documentation page.
+	 * @param {boolean} hasProperty Optional hash property to link to a specific section of the support page.
```
I understand the idea here but it won't work as expected since the selector builds a URL to our redirect endpoint on the service. The redirect has already been defined to include the fragment/anchor for the given slug, so no changes should be needed here. It's also worth noting that even if we could add the fragment here, it's better to keep the relationship abstract to allow for changes to the destination URL without requiring changes in the plugin which is one of the primary purposes of the redirect endpoint.

benbowler commented:
Thanks @aaemnnosttv, I've learned a bit more about the documentation redirect infrastructure. Reverted the `getDocumentationLinkURL` changes and updated the slug for the link.

aaemnnosttv reviewed (approved):
Thanks @benbowler 👍
